src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -314,13 +316,19 @@ private async Task RenegotiateAsync(CancellationToken cancellationToken)
                 throw new NotSupportedException(SR.Format(SR.net_io_invalidnestedcall, nameof(WriteAsync), "write"));
             }
 
+            if (_decryptedBytesCount is not 0)
+            {
+                throw new InvalidOperationException(SR.net_ssl_renegotiate_buffer);
+            }
+
             _sslAuthenticationOptions!.RemoteCertRequired = true;
-            IReadWriteAdapter adapter = new AsyncReadWriteAdapter(InnerStream, cancellationToken);
+            _isRenego = true;
```

Comment:
Where does this get set back to false? Or does it not need to?

Reply:
It does not need to but I added false back to be less confusing. According to discussion on #49346 we should allow to call NegotiateClientCertificateAsync at most once. This was already done by leaving the _nestedAuth on.